src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,33 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    if (!beast.hidden && beast instanceof Herbivore) {
```

Comment:
It's unnecessary to check for `!beast.hidden` separately from `beast instanceof Herbivore`. These conditions can be combined into a single if statement to check both that the object is an instance of Herbivore and that it is not hidden, which simplifies the logic and reduces nesting.